py3status/modules/imap.py

```diff
@@ -64,11 +67,30 @@ def check_mail(self):
 
         return response
 
+    def _connection_ssl(self):
+        connection = imaplib.IMAP4_SSL(self.imap_server,
+                                       self.port,
+                                       ssl_context=create_default_context()
+                                       )
+        return connection
+
+    def _connection_starttls(self):
+        connection = imaplib.IMAP4(self.imap_server, self.port)
+        connection.starttls(create_default_context())
+        return connection
+
     def _get_mail_count(self):
         try:
             mail_count = 0
             directories = self.mailbox.split(',')
-            connection = imaplib.IMAP4_SSL(self.imap_server, self.port)
+
+            if self.security == "ssl":
+                connection = self._connection_ssl()
+            elif self.security == "starttls":
+                connection = self._connection_starttls()
+            else:
+                raise ValueError("Unknown security protocol")
```

Comment:
Either you want to do this test in `post_config_hook()` so the module just won't run (here it will just flood the logs with tracebacks) or just test for 'starttls' else assume 'ssl'